src/Action/Order/Assign.php

```diff
@@ -6,15 +6,27 @@
 use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
 use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
 use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+use Symfony\Component\HttpFoundation\Request;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Nucleos\UserBundle\Util\CanonicalizerInterface;
 
 class Assign
 {
-    public function __construct(TokenStorageInterface $tokenStorage)
+    public function __construct(
+        TokenStorageInterface $tokenStorage,
+        RepositoryInterface $customerRepository,
+        CanonicalizerInterface $canonicalizer,
+        FactoryInterface $customerFactory
+        )
     {
         $this->tokenStorage = $tokenStorage;
+        $this->customerRepository = $customerRepository;
+        $this->canonicalizer = $canonicalizer;
+        $this->customerFactory = $customerFactory;
     }
 
-    public function __invoke($data)
+    public function __invoke($data, Request $request)
```

Comment:
Something I forgot here, is that it should check if the instance has enabled guest checkout 😕

Reply:
Fixed in 28bde80c2d44904fa527aca869fa8747390d6e4e

Reply:
Great, thanks!